### DISCUSSION TITLE
How to configure Azure Frontdoor to compress awnser from NextJS?

### DISCUSSION THREAD

**Discussion by cbou** [vercel/next.js#67961] [Help]
### Summary

Our website is under a Azure Frontdoor and NextJS ist in a Azure Container App.

We've activated compression on Frontdoor but since it doesn't have any Content-Length header, Frontdoor won't compress the response.

Now I found out that NextJS uses `Transfer-Encoding: chunked` which makes sense why the `Content-length` header isn't present.

So how can I configure Azure Frontdoor to compress awnser from NextJS? 
Or how can I prevent NextJS to use `Transfer-Encoding: chunked` and correctly set the `Content-Length` header?

**Comment by hbystrom91** [comment]
@cbou did you ever solve this? my team has migrated to the app router, and we are running into the same issue.

**Reply by cbou** [reply]
No we didn't :/ Actually you can find a lot of people online complaining how hard is it to host NextJS correctly outside of the Versel Cloud.

**Reply by hbystrom91** [reply]
@cbou thanks for your answer. this is my impression as well. we are considering cutting our losses and moving back to the pages router. the response is not streaming there, so frontdoor can handle caching and compression.

**Comment by ashleyww93** [comment]
Also running into issues where when running with Azure FrontDoor (Caching On, Compression Off), our app won't load at all. Gets stuck transferring a bundle file where only part of it is transferred.

Using Pages Router for us though, in standalone mode.

**Reply by hbystrom91** [reply]
@ashleyww93 if you do `compress: false `in the `next.config.js` and let FrontDoor handle both caching and compression, it should work. that's what we do, and our pages router pages are working fine. hope it works out for you.
